nodeup/pkg/model/node_authorizer.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"bytes"
+	"fmt"
+	"path"
+	"path/filepath"
+	"text/template"
+	"time"
+
+	"k8s.io/kops/pkg/systemd"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/nodeup/nodetasks"
+
+	"github.com/golang/glog"
+)
+
+// NodeAuthorizationBuilder is responsible for node authorization
+type NodeAuthorizationBuilder struct {
+	*NodeupModelContext
+}
+
+var _ fi.ModelBuilder = &NodeAuthorizationBuilder{}
+
+// nodeAuthorizationServiceTemplate is the template used by the node client
+var nodeAuthorizationServiceTemplate = `
+Type=oneshot
+EnvironmentFile=/etc/environment
+{{- if .client_file }}
+ExecStartPre=/usr/bin/bash -c 'while [ ! -f {{ .client_cert }} ]; do sleep 5; done; sleep 5'
+{{- end }}
+ExecStartPre=/usr/bin/mkdir -p /var/lib/kubelet
+ExecStart=/usr/bin/docker run --rm --net=host \
```

Comment:
I worry about what happens if we fail to pull the image or something like that - we won't retry.  This is fine I think for now, but for the future, one option could be that we run periodically, but have a shell script that checks for e.g. a file we create, and only runs the docker image if it isn't there.